### DISCUSSION TITLE
I am getting type error after adding interface with zod

### DISCUSSION THREAD

**Discussion by amantiwari1** [colinhacks/zod#934] [Q&A]
```ts
interface Input {
  id?: number
  select: Prisma.EnquiryFindFirstArgs["select"]
}
```

```ts
const GetEnquiry: z.ZodSchema<Input> = z.object({
  id: z.number().optional().refine(Boolean, "Required"),
  select: z.unknown(),
})
```
![image](https://user-images.githubusercontent.com/53963065/153748223-eeead3b5-cf39-440b-9f5c-822e337853c0.png)

thank you!

**Comment by JacobWeisenburger** [comment]
It seems that somewhere in `z.object()`, the type `unknown` gets converted to optional because in typescript, `undefined extends unknown`. see below:
```ts
type test1 = undefined extends unknown ? true : false
// type test1 = true
```

There might be a better way to do what you want, but I will need some info from you.
- What type is `Prisma.EnquiryFindFirstArgs["select"]`?
- What are you intending `.refine(Boolean, "Required")` to do?
